src/Nncase.Tests/Core/UnitTestTensorOfT.cs

```diff
@@ -0,0 +1,246 @@
+
+    [Fact]
+    public void TestICollection()
+    {
+        var array1 = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var momory = new Memory<int>(array1);
+        var t = new Tensor<int>(momory, new int[] { 1, 1, 2, 4 });
+        ICollection<int> c = t;
+        Assert.Equal(array1.Length, c.Count);
+        Assert.False(c.IsReadOnly);
+
+        Assert.Throws<InvalidOperationException>(() => c.Add(100));
+        Assert.Throws<InvalidOperationException>(() => c.Remove(8));
+
+        var array2 = new int[array1.Length];
+        c.CopyTo(array2, 0);
+        Assert.True(Enumerable.SequenceEqual(array1, array2));
+
+        c.Clear();
+
+        Assert.Equal(array1.Length, c.Count);
+        for (int i = 0; i < array1.Length; i++)
+        {
+            Assert.Equal(0, t.GetValue(i));
+        }
+    }
+
+    [Fact]
+    public void TestIReadOnlyCollection()
+    {
+        var length = 100;
+        var tensor = new Tensor<float>(length);
+        var a = (IReadOnlyCollection<float>)tensor;
+        Assert.Equal(length, a.Count);
+    }
+
+    [Fact]
+    public void TestIReadOnlyList()
+    {
+        var array = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var momory = new Memory<int>(array);
+        var tensor = new Tensor<int>(momory, new int[] { 8 });
+        var list = (IReadOnlyList<int>)tensor;
+        for (int i = 0; i < array.Length; i++)
+        {
+            Assert.Equal(array[i], list[i]);
+        }
+    }
+
+    [Fact]
+    public void TestIList()
+    {
+        var array = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var momory = new Memory<int>(array);
+        var tensor = new Tensor<int>(momory, new int[] { 8 });
+        var list = (IList<int>)tensor;
+
+        for (int i = 0; i < array.Length; i++)
+        {
+            list[i] = i;
+        }
+
+        for (int i = 0; i < array.Length; i++)
+        {
+            Assert.Equal(i, list[i]);
+        }
+
+        Assert.Equal(1, list.IndexOf(1));
+        Assert.Throws<InvalidOperationException>(() => list.Insert(0, 100));
+        Assert.Throws<InvalidOperationException>(() => list.RemoveAt(0));
+    }
+
+    [Fact]
+    public void TestImplicitOperator()
+    {
+        var scalar = 1F;
+        var a = (Tensor<float>)scalar;
+        Assert.Equal(scalar, a[0]);
+    }
+
+    [Fact]
+    public void TestClone()
+    {
+        var array = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var momory = new Memory<int>(array);
+        var t1 = new Tensor<int>(momory, new int[] { 8 });
+        var t2 = t1.Clone();
+        for (int i = 0; i < array.Length; i++)
+        {
+            Assert.Equal(t1[i], t2[i]);
+        }
+    }
+
+    [Fact]
+    public void TestCloneEmpty()
+    {
+        var t1 = new Tensor<float>(new int[] { 1, 2, 3, 4 });
+        var t2 = t1.CloneEmpty<int>(new int[] { 1, 3, 16, 16 });
+        var a = (ICollection<int>)t2;
+        Assert.Equal(1 * 3 * 16 * 16, a.Count);
+    }
+
+    [Fact]
+    public void TestGetValue()
```

Comment:
should test out of index for GetValue and SetValue

Reply:
ok. check throw IndexOutOfRangeException